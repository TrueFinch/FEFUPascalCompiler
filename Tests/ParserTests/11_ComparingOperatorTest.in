program a;
var
    i, k: integer;
begin
    i := 4;
    k := 2;
    if (i > k) then
        write(i);
    if (k < i) then
        writeln(k);
    if (k <> i) then 
        writeln(i, k);
    if (i = 4) then
        write(i);
    if (k <= i) then
        writeln(k);
    if (i >= k) then
        writeln(i, k);
end.