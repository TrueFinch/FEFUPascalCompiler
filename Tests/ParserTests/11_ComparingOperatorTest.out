                                                                          a                                                                          
                                                                          ┼                                                                          
                                                                      MainBlock                                                                      
       ┌──────────────────────────────────────────────────────────────────┴──────┐                                                                   
      var                                                                CompoundStatement                                                           
       ┼         ┌─────┬────────────┬───────────────────┬────────────────────┬───┴───────────────┬───────────────────┬────────────────────┐          
 SimpleVarDecl   :=    :=      IfStatement         IfStatement          IfStatement         IfStatement         IfStatement          IfStatement     
 ┌──┬──┴──┐     ┌┴─┐  ┌┴─┐   ┌──────┴──┐         ┌──────┴──┐         ┌───────┴──┐         ┌──────┴──┐         ┌──────┴──┐         ┌───────┴──┐       
 i  k  integer  i  4  k  2   >    FunctionCall   <    FunctionCall   <>   FunctionCall    =    FunctionCall   <=   FunctionCall   >=   FunctionCall  
                            ┌┴─┐    ┌──┴─┐      ┌┴─┐     ┌─┴───┐    ┌┴─┐     ┌──┴──┬──┐  ┌┴─┐    ┌──┴─┐      ┌┴─┐     ┌─┴───┐    ┌┴─┐     ┌──┴──┬──┐ 
                            i  k  write  i      k  i  writeln  k    k  i  writeln  i  k  i  4  write  i      k  i  writeln  k    i  k  writeln  i  k 